sql/semisync_master.cc

```diff
@@ -641,6 +612,13 @@ int Repl_semi_sync_master::report_reply_packet(uint32 server_id,
 
   rpl_semi_sync_master_get_ack++;
   report_reply_binlog(server_id, log_file_name, log_file_pos);
+  DBUG_EXECUTE_IF("sync_semisync_report_reply",
+    debug_sync_set_action(current_thd, STRING_WITH_LEN(
+      /*FIXME: no DBUG_ASSERT due to
+        MDEV-32748 debug_sync_set_action() Fails if Entered in an Error State */
+      "now SIGNAL report_reply_wait WAIT_FOR report_reply_continue"
+    ));
+  );
```

Comment:
Oops, I placed the comment on the wrong line.
```suggestion
  /*FIXME: no DBUG_ASSERT due to
    MDEV-32748 debug_sync_set_action() Fails if Entered in an Error State */
  DBUG_EXECUTE_IF("sync_semisync_report_reply",
    debug_sync_set_action(current_thd, STRING_WITH_LEN(
      "now SIGNAL report_reply_wait WAIT_FOR report_reply_continue"
    ));
  );
```

But really, we should figure out what happened to the ACK receiver thread that it managed to scrape up a handled
“Got an error reading communication packets” and lingers it in `thd->is_error()`.